src/databricks/labs/ucx/source_code/dependencies.py

```diff
@@ -15,7 +15,7 @@ def from_object_info(object_info: ObjectInfo):
         assert object_info.path is not None
         return Dependency(object_info.object_type, object_info.path)
 
-    def __init__(self, object_type: ObjectType, path: str):
+    def __init__(self, object_type: ObjectType | None, path: str):
```

Comment:
We need `ObjectType | None` because python import statements don't have an ObjectType until resolved

Reply:
it'll always be FILE

Reply:
No, `__init__` is called at discovery time, we don't know if it's a FILE until resolved. It could be a builtin or pre-installed package. Making it a FILE in these cases is misleading and error prone. See `dependencies.py`, lines 140-142 in PR#1367   
```
resolved = self._resolver.resolve(dependency)
if resolved is None:
    return None
```